### PR TITLE
Make `shelfmark` a required field on source create and edit forms

### LINKED ISSUE #1604: TypeError on Source Create page
Try and create a source with all of the fields empty. We get the following error:

`TypeError: sequence item 0: expected str instance, NoneType found`

This is currently on the develop and staging branches.

### PR DESCRIPTION
**Pull request by lucasmarchd01**
This was causing a TypeError on the source create and edit forms. Also adds the fix to the admin source create form. Fixes #1604

### CODE REVIEW

**Comment by ahankinson** [comment]
Shelfmark should also be marked as required in the `Source` model. Also, `holding_institution` needs to be marked as required in both the editing form and the model.

**Comment by dchiller** [comment]
> Shelfmark should also be marked as required in the `Source` model. Also, `holding_institution` needs to be marked as required in both the editing form and the model.

Was the fact that these weren't previously required because null needed to be allowed while we transitioned the records? But the intention was that they really shouldn't allow null after that?

**Comment by ahankinson** [comment]
> Was the fact that these weren't previously required because null needed to be allowed while we transitioned the records? 

Yep.

**Comment by dchiller** [comment]
Great... I think it makes sense to make those changes, and then when we update production we just won't apply this migration, then run `migrate_records`, and then apply this last migration.

@lucasmarchd01 Do you want to add those changes here or we can create another issue?

**Comment by lucasmarchd01** [comment]
> @lucasmarchd01 Do you want to add those changes here or we can create another issue?

Sure, we can add those changes here.

File: django/cantusdb_project/main_app/templates/source_create.html [hunk 1]
```diff
@@ -57,7 +57,7 @@ <h3>Create Source</h3>
                 <div class="form-row">
                     <div class="form-group m-1 col-lg-3">
                         <label for="{{ form.shelfmark.id_for_label }}">
-                            <small>Shelfmark:</small>
+                            <small>Shelfmark:</small><span class="text-danger" title="This field is required">*</span></small>
```
Comment (ahankinson) [inline django/cantusdb_project/main_app/templates/source_create.html]: See: https://github.com/DDMAL/CantusDB/issues/1550

Better would be to add the `.small` class to the `<label>` element.

**Comment by lucasmarchd01** [comment]
There's a migration script that needs to be added, but I can generate that after #1615 is merged.

File: django/cantusdb_project/main_app/templates/source_create.html (hunk 1 above)
Comment (lucasmarchd01) [reply django/cantusdb_project/main_app/templates/source_create.html]: I think I've fixes the ones for these fields by using `class='form-control-sm` on the `<label>` element. But there's many more of these cases we'll want to fix in the future.

**Review by dchiller — DISMISSED** [review]
Cool. Good to merge imo with the migration added :)

**Review by dchiller — APPROVED** [review]

**Pull request merged**